### PR TITLE
[refactor] multiprocess_pipe: cleanup __init__

### PR DESCRIPTION
**Pull request by msbaines**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: fairscale/nn/pipe/multiprocess_pipe.py [hunk 1]
```diff
@@ -259,32 +256,29 @@ def __init__(
...
-            self._skip_layout = SkipLayout(len(module), {})  # FIXME(tom)
+            self.create_pipeline()
 
-        rank = self.group.rank()
-        if rank >= len(self.balance):
-            self.pipeline = None
-            self.final_stage = False
-        else:
-            self.final_stage = rank == len(self.balance) - 1
+        del module
```
Comment (min-xu-ai) [inline fairscale/nn/pipe/multiprocess_pipe.py:280]: add a comment why is this needed?

File: fairscale/nn/pipe/multiprocess_pipe.py [hunk 2]
```diff
@@ -259,32 +256,29 @@ def __init__(
...
+        if isinstance(module, nn.Sequential):
+            local_partitions = split_module(module, self.balance)
+            self._skip_layout = inspect_skip_layout(local_partitions)
+        else:
+            self._skip_layout = SkipLayout(len(module), {})  # FIXME(tom)
+
         rank = self.group.rank()
+        self.final_stage = rank == len(self.balance) - 1
         if rank >= len(self.balance):
             warnings.warn("More ranks than partitions, some ranks unused")
```
Comment (min-xu-ai) [inline fairscale/nn/pipe/multiprocess_pipe.py:268]: do we really need to support this case with code coverage, unit test etc or should just assert fail?

**Review by min-xu-ai — APPROVED** [review]
LGTM

File: fairscale/nn/pipe/multiprocess_pipe.py [hunk 3]
```diff
@@ -220,9 +220,6 @@ def __init__(
     ) -> None:
         super().__init__()
 
-        chunks = int(chunks)
-        checkpoint = str(checkpoint)
```
Comment (sidgoyal78) [inline fairscale/nn/pipe/multiprocess_pipe.py:224]: this is failing

**Review by sidgoyal78 — APPROVED** [review]
looks good, I guess you already saw the failing test.

File: fairscale/nn/pipe/multiprocess_pipe.py (hunk 2 above)
Comment (msbaines) [reply fairscale/nn/pipe/multiprocess_pipe.py:268]: Good point. Could would be simpler if we just remove this and error.
